db/Models/app.models.js

```diff
@@ -0,0 +1,18 @@
+const db = require("../connection");
+
+function fetchCategories() {
+  return db
+    .query(
+      `
+    SELECT * FROM categories;
+    `
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject("could not find this data");
```

Comment:
The combo of this and the error controller you have are not quite correct as we usually check the status code and go from there